core/commands/log.py

```diff
@@ -56,12 +56,42 @@ def run_async(self):
                 "Skip this set of commits and choose from the next-oldest batch."
             ])
 
+        try:
+            pre_selected_index = self._hashes.index(self._selected_hash) if hasattr(self, '_selected_hash') else 0
+        except ValueError:
+            pre_selected_index = 0
+
+        # _on_hash_selection has to be called by set_timeout_async
+        # otherwise, on_highlight_commit_async will be called after
+        # _on_hash_selection is executed.
         self.window.show_quick_panel(
             self._entries,
-            self.on_hash_selection,
-            flags=sublime.MONOSPACE_FONT
+            lambda index: sublime.set_timeout_async(lambda: self._on_hash_selection(index), 10),
```

Comment:
To make sure it runs after `on_highlight_commit`. It seems that both `on_highlight_commit` and `_on_hash_selection` are executed when enter is hit